Drop-in compatability with itzg/minecraft-server on Docker hub & minor cleanup

morgaesis opened the pull request:
- [x] Maintain backwards compatibility with `MEMORYSIZE`
- [ ] ~Use user `1000` instead of non-standard `9001`~
- [x] Use same environment variables as itzg/minecraft-server (see [their docs](https://docker-minecraft-server.readthedocs.io/en/latest/variables/))
- [x] Various formatting
- [x] Fix broken badge in README

morgaesis committed:
Update Dockerfile

morgaesis committed:
Update docker-entrypoint.sh

morgaesis committed:
Formatting

morgaesis committed:
No args to badge

morgaesis committed:
No MEMORYSIZE

mtoensing commented:
Okay, thanks! I understand that you would like to establish a compatibility with the more popular itzg image to one can change the container more easily. That means that min and max memory are split into to variables. I get it. 

May I ask why you removed this: 

```
# Volumes for the external data (Server, World, Config...)
VOLUME "/data"
```

1. Isn't this volume parameter needed? 
4. Will the change of the user Id be backwards compatible? 

Let's merge this, but I need to understand it. =)

morgaesis committed:
Re-add VOLUME directive in Dockerfile

morgaesis committed:
Formatting Dockerfile

morgaesis committed:
Formatting entrypoint

morgaesis committed:
Revert UID/GID to 9001

morgaesis commented:
Thanks for the quick review and reply :heart: 

>     1. Isn't this volume parameter needed?
Yes, if we want the volume to be created in all cases. I assumed that the user would always use `-v` for persistence, but this prevents accidentally deleting worlds that would only reside _inside_ the container. Reverted.

>     2. Will the change of the user Id be backwards compatible?
Yes, it would break backwards compatibility. Although I disagree with the UID choice to begin with, backwards compatibility is more important. Reverted, and PR description updated.

Styne13 commented on Dockerfile:
```diff
@@ -24,11 +24,11 @@ FROM eclipse-temurin:21-jre AS runtime
 ARG TARGETARCH
```
`
ARG version=1.20.4
`
Maybe also bump version to 1.20.6?

mtoensing replied:
No yet. It is still experimental. https://papermc.io/downloads/paper

mtoensing commented:
Sorry, had to revert. I tried to add backwards compatibility to memorysize parameters myself but was not successful. Please make sure to add backwards compatibility. And please leave the docs at they are. I dont really see the point to confuse anyone with a new memory parameter in the documentation
